src/Quahog/Result.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Xenolope\Quahog;
+
+class Result
+{
+    private const RESULT_OK    = 'OK';
+    private const RESULT_FOUND = 'FOUND';
+    private const RESULT_ERROR = 'ERROR';
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string|null
+     */
+    private $reason;
+
+    /**
+     * @var string
+     */
+    private $status;
+
+    public function __construct(string $status, string $filename, ?string $reason, ?string $id)
+    {
+        $this->status   = $status;
+        $this->filename = $filename;
+        $this->reason   = $reason;
+        $this->id       = $id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId(): ?string
```

Comment:
Could you add docblocks for all the public methods on this class? A short description is fine, as long as it's clear.

Reply:
I see there's also an inconsistency in the annotation of `getId()` :smile:. I'll add them.